src/NuGet.Core/NuGet.Commands/RestoreCommand/LockFileBuilder.cs

```diff
@@ -265,6 +269,43 @@ public class LockFileBuilder
                                 librariesWithWarnings.Add(library);
                             }
                         }
+
+                        if (warnFor30FrameworkRefs && targetLibrary.FrameworkReferences.Any())
+                        {
+                            bool referencedMicrosoftWindowsDesktopApp = false;
+
+                            foreach (var frameworkReference in targetLibrary.FrameworkReferences)
+                            {
+                                if (frameworkReference.StartsWith("Microsoft.WindowsDesktop.App|"))
```

Comment:
you like three strings enough to avoid the startswith?

private static readonly string windowsDesktopApp = "Microsoft.WindowsDesktop.App";
private static readonly string windowsDesktopAppWPF = "Microsoft.WindowsDesktop.App|WPF";
private static readonly string windowsDesktopAppWinforms = "Microsoft.WindowsDesktop.App|WinForms";
...
if (StringComparer.Equals(windowsDesktopAppWPF, frameworkReference)
 || StringComparer.Equals(windowsDesktopAppWinforms, frameworkReference))
...
var message = string.Format(CultureInfo.CurrentCulture,
    Strings.Log_UsingWindowsDesktopFrameworkRefWarning,
    libraryName,
    nonFallbackFramework,
    windowsDesktopApp);

Reply:
I prefer 3 strings that can be reused from hereon, rather than the 6 adhoc strings you are committing. However, it's not a huge deal, just a smell.

Reply:
per other resolution...this code is being removed. so resolving conversation.